Libraries/xcdriver/Sources/Usage.cpp

```diff
@@ -0,0 +1,85 @@
+/**
+ Copyright (c) 2016-present, Facebook, Inc.
+ All rights reserved.
+
+ This source code is licensed under the BSD-style license found in the
+ LICENSE file in the root directory of this source tree. An additional grant
+ of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#include <xcdriver/Usage.h>
+#include <libutil/FSUtil.h>
+#include <libutil/SysUtil.h>
+#include <sstream>
+
+using libutil::FSUtil;
+using libutil::SysUtil;
+
+std::string xcdriver::usage() {
+	std::string executable = FSUtil::GetBaseName(SysUtil::GetExecutablePath());
+
+	std::ostringstream result;
```

Comment:
nit: these lines seem a little extra indented.

Also, in C++, `std::string` is mutable. Combined with how operator overloading interacts with references, that means that `+=` isn't any less efficient than using a string builder / string stream. That's unlike Java or Objective-C, where `StringBuilder` or `NSMutableString` would be more efficient than appending immutable strings repeatedly.